pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +20,27 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	azureDNS, err := azureconfig.NewDNSConfig()
+	if err != nil {
+		return err
+	}
+
+	rgName := installConfig.Config.Azure.BaseDomainResourceGroupName
+	zoneName := installConfig.Config.BaseDomain
```

Comment:
only external clusters need access to the public DNS, so we should skip all of this when it is not External

Reply:
Agreed, which brings up for question: how do we determine an existing private cluster exists? I guess we do the same sort of DNS check, but using private DNS.

Reply:
I think for internal only clusters we don't need this validation at all... because we create private zone in a unique resource group. So using same name will not affect existing clusters.

Reply:
Fixed.